src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemFinalStateValidator.cs

```diff
@@ -0,0 +1,127 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using FellowOakDicom;
+using Microsoft.Health.Dicom.Core.Extensions;
+using Microsoft.Health.Dicom.Core.Features.Workitem.Model;
+
+namespace Microsoft.Health.Dicom.Core.Features.Workitem
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    internal static class WorkitemFinalStateValidatorExtension
+    {
+        private static readonly HashSet<FinalStateRequirementDetail> Requirements = GetRequirements();
+
+        public static void ValidateFinalStateRequirement(this DicomDataset dataset)
+        {
+            var procedureStepState = dataset.GetProcedureState();
+
+            foreach (var requirement in Requirements)
+            {
+                dataset.ValidateRequirement(requirement.DicomTag, procedureStepState, requirement.RequirementCode);
+
+                if (null != requirement.SequenceRequirements)
+                {
+                    dataset.ValidateSequence(requirement.DicomTag, procedureStepState, requirement.SequenceRequirements);
+                }
+            }
+        }
+
+        private static void ValidateSequence(this DicomDataset dataset, DicomTag sequenceTag, ProcedureStepState procedureStepState, IReadOnlyCollection<FinalStateRequirementDetail> requirements)
+        {
+            if (requirements.Count == 0)
+            {
+                return;
+            }
+
+            // TODO: Should we consider throwing an exception here?
+            if (!dataset.TryGetSequence(sequenceTag, out var sequence))
+            {
+                return;
+            }
+
+            var sequenceDatasets = sequence.Items;
+
+            // TODO: Should we just search for the requirements in the 1st Dataset in the Sequence?
```

Comment:
@josiahvinson Should we just search for the requirements in the 1st Dataset in the Sequence?